client/client.go

```diff
@@ -383,25 +393,45 @@ func (r *NotaryRepository) Publish() error {
 		return err
 	}
 
+	// these are the tuf files we will need to update, serialized as JSON before
+	// we send anything to remote
+	update := make(map[string][]byte)
+
 	// check if our root file is nearing expiry. Resign if it is.
-	if nearExpiry(r.tufRepo.Root) || r.tufRepo.Root.Dirty {
-		if err != nil {
-			return err
-		}
+	if nearExpiry(r.tufRepo.Root) || r.tufRepo.Root.Dirty || updateRoot {
 		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"))
+
+		rootJSON, err := json.Marshal(root)
 		if err != nil {
 			return err
 		}
-		updateRoot = true
+		update[data.CanonicalRootRole] = rootJSON
 	}
-	// we will always resign targets and snapshots
-	targets, err := r.tufRepo.SignTargets("targets", data.DefaultExpires("targets"))
+
+	// we will always resign targets, and snapshots if we have the snapshots key
+	targetsJSON, err := serializeCanonicalRole(r.tufRepo, data.CanonicalTargetsRole)
```

Comment:
We're getting targetsJSON as the output of `serializeCanonicalRole` when 10 lines up we're getting rootJSON, as the output of a direct call to `json.Marshal(root)` where `root` comes from a direct `r.tufRepo.SignRoot`. We should have a single way of getting the resigned files for all roles instead.